packages/taiko-client/prover/proof_submitter/proof_submitter.go

```diff
@@ -198,21 +201,30 @@ func (s *ProofSubmitter) RequestProof(ctx context.Context, meta metadata.TaikoBl
 				}
 				return fmt.Errorf("failed to request proof (id: %d): %w", meta.GetBlockID(), err)
 			}
-			if meta.IsOntakeBlock() && s.proofBuffer.MaxLength > 1 {
+			if meta.IsOntakeBlock() && s.proofBuffer.Enabled() {
 				bufferSize, err := s.proofBuffer.Write(result)
 				if err != nil {
-					return fmt.Errorf("failed to add proof into buffer (id: %d)(current buffer size: %d): %w",
+					return fmt.Errorf(
+						"failed to add proof into buffer (id: %d) (current buffer size: %d): %w",
 						meta.GetBlockID(),
 						bufferSize,
 						err,
 					)
 				}
-				log.Debug("Succeed to generate proof",
+				log.Info(
+					"Proof generated",
 					"blockID", meta.GetBlockID(),
 					"bufferSize", bufferSize,
+					"maxBufferSize", s.proofBuffer.MaxLength,
+					"bufferIsAggregating", s.proofBuffer.IsAggregating(),
+					"bufferLastUpdatedAt", s.proofBuffer.lastUpdatedAt,
 				)
-				if s.proofBuffer.MaxLength == uint64(bufferSize) {
+				// Check if we need to aggregate proofs.
+				if !s.proofBuffer.IsAggregating() &&
+					(uint64(bufferSize) >= s.proofBuffer.MaxLength ||
+						time.Since(s.proofBuffer.lastUpdatedAt) > s.forceBatchProvingInterval) {
```

Comment:
https://github.com/taikoxyz/taiko-mono/blob/3a06487b2c5b91489c96baac29289158f56462b2/packages/taiko-client/prover/proof_submitter/proof_buffer.go#L109
There exists such a case, if an existing proof A is detected in the buffer, here it will wait for a `forceBatchProvingInterval` before submitting the aggregation, but during this waiting period the proof B in the buffer expires and is proved by someone else, and after clearing the B it needs to wait for a `forceBatchProvingInterval` to submit the aggregation, but C...this will fall into an infinite waiting situation.

In a rare case, when we've accumulated a few proofs and the buffer is not full, unfortunately, we were never again assigned as the prover, we would never submit the aggregation although reached the `forceBatchProvingInterval`

Reply:
will https://github.com/taikoxyz/taiko-mono/pull/18630 mitigate this issue?